### ISSUE TITLE
failed when using yarn script

### ISSUE THREAD

**Issue by alexeikaratai** [eslint/eslint#10037]
**Tell us about your environment**

* **ESLint Version:** 4.18.1
* **Node Version:** 9.5.0
* **npm Version:** 5.6.1

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
{
  "extends": [
    "prettier",
    "airbnb"
  ],
  "env": {
    "es6": true,
    "node": true,
    "browser": true
  },
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules":{
    "linebreak-style": [0, "windows"], // dif \n osx vs win
    "react/jsx-filename-extension": [2, {
      "extensions": [".js"]
    }],
    "react/prop-types": 0,
    "react/jsx-no-bind": [2, {
      "ignoreRefs": false,
      "allowArrowFunctions": false,
      "allowBind": false
    }],
    "react/forbid-prop-types": [2, { "forbid": ["any", "array"] }],
    "jsx-a11y/href-no-hash": "off",
    "jsx-a11y/label-has-for": "off",
    "no-mixed-operators": ["off"],
    "no-underscore-dangle":"off"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash
**work** ./node_modules/.bin/eslint -c .eslintrc --fix src/ 
```
**DONT work** yarn eslint where "eslint": "./node_modules/.bin/eslint -c .eslintrc --fix src/",
**What did you expect to happen?**

**What actually happened? Please include the actual, raw output from ESLint.**

yarn run v1.5.1
$ ./node_modules/.bin/eslint -c .eslintrc --fix src/

/Users/alexkaratai/work/demo/src/components/generators.js
  14:12  warning  Unexpected console statement  no-console
  19:12  warning  Unexpected console statement  no-console

/Users/alexkaratai/work/demo/src/containers/Create.js
  30:25  warning  Unexpected console statement  no-console
  31:27  warning  Unexpected console statement  no-console
  43:21  warning  Unexpected console statement  no-console
  44:23  warning  Unexpected console statement  no-console

/Users/alexkaratai/work/demo/src/containers/Login.js
  28:25  warning  Unexpected console statement  no-console
  34:11  warning  Unexpected alert              no-alert
  36:11  warning  Unexpected alert              no-alert

/Users/alexkaratai/work/demo/src/containers/Main.js
  13:8  error  'GeneratorsMain' is defined but never used  no-unused-vars

✖ 10 problems (1 error, 9 warnings)

error An unexpected error occurred: "Command failed.
Exit code: 1
Command: sh
Arguments: -c ./node_modules/.bin/eslint -c .eslintrc --fix src/
Directory: /Users/alexkaratai/work/demo
Output:
".
info If you think this is a bug, please open a bug report with the information provided in "/Users/alexkaratai/work/demo/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by platinumazure** [comment]
Hi @alexeikaratai, thanks for the issue.

I don't follow. What did you expect to happen? Since you haven't included any code samples, I don't know if you're expecting the code to be lint-free, or if you are expecting some other result. Please fill in all of the questions we ask in our issue template. Thanks!

**Comment by Ajdija** [comment]
@platinumazure @not-an-aardvark please reopen, I have the same issue and it works from the command line:
```
 ./node_modules/eslint/bin/eslint.js  src
```
but not from the package.json script:
```
    "lint": "./node_modules/eslint/bin/eslint.js src"
```
(yarn run lint)

---
file contents to lint do not make the difference,
package.json script run full error:
```
$ yarn run lint
yarn run v1.5.1
$ ./node_modules/eslint/bin/eslint.js src

<local_path>/<project>/src/LoginForm.js
   2:8   error    A space is required after '{'                 object-curly-spacing
   ...more errors...

✖ 16 problems (14 errors, 2 warnings)
  10 errors, 0 warnings potentially fixable with the `--fix` option.

error An unexpected error occurred: "Command failed.
Exit code: 1
Command: sh
Arguments: -c ./node_modules/eslint/bin/eslint.js src
Directory: <local_path>/<project>/
Output:
".
info If you think this is a bug, please open a bug report with the information provided in "<local_path>/<project>/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
while command line run full error (all fine):
```
$ ./node_modules/eslint/bin/eslint.js  src

<local_path>/<project>/src/LoginForm.js
   2:8   error    A space is required after '{'                 object-curly-spacing
   ...more errors...

✖ 16 problems (14 errors, 2 warnings)
  10 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by platinumazure** [comment]
@Ajdija Please open a new issue, thanks!
